src/ApiGateways/Web.Bff.Shopping/aggregator/Infrastructure/HttpClientAuthorizationDelegatingHandler.cs

```diff
@@ -1,49 +1,40 @@
-﻿using Microsoft.AspNetCore.Authentication;
-using Microsoft.AspNetCore.Http;
-using System.Collections.Generic;
-using System.Net.Http;
-using System.Net.Http.Headers;
-using System.Threading;
-using System.Threading.Tasks;
-
-namespace Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Infrastructure
+﻿namespace Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Infrastructure;
+
+public class HttpClientAuthorizationDelegatingHandler
+        : DelegatingHandler
 {
-    public class HttpClientAuthorizationDelegatingHandler
-         : DelegatingHandler
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public HttpClientAuthorizationDelegatingHandler(IHttpContextAccessor httpContextAccessor)
     {
-        private readonly IHttpContextAccessor _httpContextAccessor;
+        _httpContextAccessor = httpContextAccessor;
+    }
 
-        public HttpClientAuthorizationDelegatingHandler(IHttpContextAccessor httpContextAccessor)
-        {
-            _httpContextAccessor = httpContextAccessor;
-        }
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        var authorizationHeader = _httpContextAccessor.HttpContext
+            .Request.Headers["Authorization"];
 
-        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        if (!string.IsNullOrEmpty(authorizationHeader))
         {
-            var authorizationHeader = _httpContextAccessor.HttpContext
-                .Request.Headers["Authorization"];
-
-            if (!string.IsNullOrEmpty(authorizationHeader))
-            {
-                request.Headers.Add("Authorization", new List<string>() { authorizationHeader });
-            }
-
-            var token = await GetToken();
+            request.Headers.Add("Authorization", new List<string>() { authorizationHeader });
+        }
 
-            if (token != null)
-            {
-                request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", token);
-            }
+        var token = await GetToken();
 
-            return await base.SendAsync(request, cancellationToken);
+        if (token != null)
+        {
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", token);
         }
 
-        async Task<string> GetToken()
-        {
-            const string ACCESS_TOKEN = "access_token";
+        return await base.SendAsync(request, cancellationToken);
+    }
 
-            return await _httpContextAccessor.HttpContext
-                .GetTokenAsync(ACCESS_TOKEN);
-        }
+    async Task<string> GetToken()
+    {
+        const string ACCESS_TOKEN = "access_token";
+
+        return await _httpContextAccessor.HttpContext
+            .GetTokenAsync(ACCESS_TOKEN);
     }
```

Comment:
```suggestion
    Task<string> GetTokenAsync()
    {
        const string ACCESS_TOKEN = "access_token";

        return _httpContextAccessor.HttpContext
            .GetTokenAsync(ACCESS_TOKEN);
    }
```

It's awaited where it is consumed, we can avoid an extra state-machine step by just returning the `Task` here that represents the asynchronous operation - as the result is not used here.